rviz_common/src/rviz_common/interaction/view_picker.cpp

```diff
@@ -171,9 +171,9 @@ void ViewPicker::getPatchDepthImage(
   auto data_ptr = static_cast<uint8_t *>(depth_pixel_box_.data);
 
   for (uint32_t pixel = 0; pixel < num_pixels; ++pixel) {
-    uint8_t a = data_ptr[4 * pixel];
-    uint8_t b = data_ptr[4 * pixel + 1];
-    uint8_t c = data_ptr[4 * pixel + 2];
+    uint8_t a = data_ptr[3 * pixel];
+    uint8_t b = data_ptr[3 * pixel + 1];
+    uint8_t c = data_ptr[3 * pixel + 2];
```

Comment:
Interesting. I took a look through the code, and I *think* we expect three values per pixel because of the format set here:

https://github.com/ros2/rviz/blob/3ca50ac6a983dc3cf6fa3690fa93890a7983d580/rviz_common/src/rviz_common/interaction/view_picker.cpp#L234

It's possible in ROS 1, the format included an alpha channel, hence 4 bytes. It might be a good idea to assert that the pixel box data is three-channel, prior to extracting values. Could you try adding the following before this loop?

```c++
assert(Ogre::PF_R8G8B8 == depth_pixel_box_.format);
```

It might be worth leaving that in there just in case, or at least referencing in a comment where this format is coming from.

Reply:
I added this assert and an explanatory comment in https://github.com/ros2/rviz/pull/661

Reply:
I was trying to figure out _where_ this actually changed as well, unfortunately without much success so far. The `depth_render_texture_` is initialized with the same `PF_R8G8B8` enum value in [Melodic](https://github.com/ros-visualization/rviz/blob/e51648b148ca9bdb71ee45cd569dfc24e37969a6/src/rviz/selection/selection_manager.cpp#L364-L366) and [Noetic](
https://github.com/ros-visualization/rviz/blob/ba536b2d5baae4be167f9741a3f5bcb05d129af6/src/rviz/selection/selection_manager.cpp#L365-L367) RViz:

```
depth_render_texture_ = Ogre::TextureManager::getSingleton().createManual(
  tex_name, Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME, Ogre::TEX_TYPE_2D,
  depth_texture_width_, depth_texture_height_, 0, Ogre::PF_R8G8B8, Ogre::TU_RENDERTARGET);
```